api/codelists.go

```diff
@@ -14,13 +16,32 @@ func (c *CodeListAPI) getCodeLists(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	filterBy := r.URL.Query().Get("type")
 
+	// get limit from query parameters, or default value
+	limit, err := utils.GetPositiveIntQueryParameter(r.URL.Query(), "limit", c.defaultLimit)
```

Comment:
These two blocks of code to extract the limit and offset are repeated for each endpoint. Consider extracting a single function into `api.go` that all endpoints can use.